<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>UiKeyExchange</class>
 <widget class="QWidget" name="UiKeyExchange">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>945</width>
    <height>559</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>KeyExchange</string>
  </property>
  <layout class="QGridLayout" name="gridLayout">
   <property name="leftMargin">
    <number>0</number>
   </property>
   <property name="topMargin">
    <number>0</number>
   </property>
   <property name="rightMargin">
    <number>0</number>
   </property>
   <property name="bottomMargin">
    <number>0</number>
   </property>
   <item row="0" column="0">
    <widget class="QTabWidget" name="keyExchange_tabWidget_X25519">
     <property name="currentIndex">
      <number>0</number>
     </property>
     <widget class="QWidget" name="tabName_KeyExchange_X25519">
      <attribute name="title">
       <string>X25519</string>
      </attribute>
      <layout class="QGridLayout" name="gridLayout_7">
       <item row="0" column="0">
        <widget class="QGroupBox" name="groupBox_KeysGenAndSharedCalc">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="minimumSize">
          <size>
           <width>0</width>
           <height>242</height>
          </size>
         </property>
         <property name="title">
          <string>Key pairs gen and shared secret calc</string>
         </property>
         <layout class="QGridLayout" name="gridLayout_4">
          <item row="0" column="0">
           <widget class="QLabel" name="x25519_label_PrivateKey">
            <property name="text">
             <string notr="true">Own private key (32 bytes)</string>
            </property>
           </widget>
          </item>
          <item row="3" column="0" colspan="2">
           <widget class="QLineEdit" name="x25529_textEdit_ownPublicKey">
            <property name="font">
             <font>
              <family>Consolas</family>
             </font>
            </property>
            <property name="text">
             <string>c8 a0 c8 23 86 7f 9c 1f 19 14 c6 ee aa 40 20 32 c3 16 ad af 0e 24 74 64 98 ba 64 a3 c0 6c 15 6c</string>
            </property>
            <property name="clearButtonEnabled">
             <bool>true</bool>
            </property>
           </widget>
          </item>
          <item row="2" column="0">
           <widget class="QLabel" name="x25519_label_PublicKey">
            <property name="text">
             <string>Own public key (32 bytes)</string>
            </property>
           </widget>
          </item>
          <item row="3" column="2" colspan="2">
           <widget class="QLineEdit" name="x25529_textEdit_otherPublicKey">
            <property name="font">
             <font>
              <family>Consolas</family>
             </font>
            </property>
            <property name="text">
             <string>85 20 F0 09 89 30 A7 54 74 8B 7D DC B4 3E F7 5A 0D BF 3A 0D 26 38 1A F4 EB A4 A9 8E AA 9B 4E 6A</string>
            </property>
            <property name="clearButtonEnabled">
             <bool>true</bool>
            </property>
           </widget>
          </item>
          <item row="6" column="3">
           <widget class="QPushButton" name="x25519_pushButton_Clear">
            <property name="text">
             <string>Clear </string>
            </property>
           </widget>
          </item>
          <item row="4" column="0">
           <widget class="QLabel" name="x25519_label_ownSharedSecret">
            <property name="text">
             <string>Own calculated shared secret</string>
            </property>
           </widget>
          </item>
          <item row="0" column="2">
           <widget class="QLabel" name="x25519_label_PrivateKey_2">
            <property name="text">
             <string notr="true">Other private key (32 bytes)</string>
            </property>
           </widget>
          </item>
          <item row="5" column="0" colspan="2">
           <widget class="QLineEdit" name="x25529_textEdit_ownSharedSecret">
            <property name="font">
             <font>
              <family>Consolas</family>
             </font>
            </property>
            <property name="clearButtonEnabled">
             <bool>true</bool>
            </property>
           </widget>
          </item>
          <item row="4" column="2">
           <widget class="QLabel" name="x25519_label_PublicKey_3">
            <property name="text">
             <string>Other calculated shared secret</string>
            </property>
           </widget>
          </item>
          <item row="1" column="2" colspan="2">
           <widget class="QLineEdit" name="x25529_textEdit_otherPrivateKey">
            <property name="font">
             <font>
              <family>Consolas</family>
             </font>
            </property>
            <property name="text">
             <string>77 07 6D 0A 73 18 A5 7D 3C 16 C1 72 51 B2 66 45 DF 4C 2F 87 EB C0 99 2A B1 77 FB A5 1D B9 2C 2A</string>
            </property>
            <property name="cursorPosition">
             <number>95</number>
            </property>
            <property name="clearButtonEnabled">
             <bool>true</bool>
            </property>
           </widget>
          </item>
          <item row="6" column="0">
           <widget class="QPushButton" name="x25519_pushButton_GenerateEphemeralKeys">
            <property name="text">
             <string>Generate ephemeral keys pair</string>
            </property>
           </widget>
          </item>
          <item row="5" column="2" colspan="2">
           <widget class="QLineEdit" name="x25529_textEdit_otherSharedSecret">
            <property name="font">
             <font>
              <family>Consolas</family>
             </font>
            </property>
            <property name="clearButtonEnabled">
             <bool>true</bool>
            </property>
           </widget>
          </item>
          <item row="2" column="2">
           <widget class="QLabel" name="x25519_label_PublicKey_4">
            <property name="text">
             <string>Other public key (32 bytes)</string>
            </property>
           </widget>
          </item>
          <item row="6" column="1">
           <widget class="QPushButton" name="x25519_pushButton_CalculateSharedSecret">
            <property name="text">
             <string>Calculate shared secret</string>
            </property>
           </widget>
          </item>
          <item row="1" column="0" colspan="2">
           <widget class="QLineEdit" name="x25529_textEdit_ownPrivateKey">
            <property name="font">
             <font>
              <family>Consolas</family>
             </font>
            </property>
            <property name="text">
             <string>30 23 27 a9 07 bd 94 aa ff d5 7b b2 d9 97 1f a0 4c 7f df cd e5 3f 6f 9b 2b b7 2a 4f 22 95 b0 4c</string>
            </property>
            <property name="clearButtonEnabled">
             <bool>true</bool>
            </property>
           </widget>
          </item>
          <item row="7" column="0" colspan="4">
           <widget class="QLabel" name="label_2">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="text">
             <string>Left (Own private key + other public key), Right (other private key + own public key).
 ** Private key can be preset and only public key will be calculated.</string>
            </property>
            <property name="alignment">
             <set>Qt::AlignCenter</set>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item row="1" column="0">
        <widget class="QGroupBox" name="groupBox_SymKeyCalc">
         <property name="minimumSize">
          <size>
           <width>0</width>
           <height>201</height>
          </size>
         </property>
         <property name="font">
          <font>
           <bold>false</bold>
          </font>
         </property>
         <property name="title">
          <string>Symmetric keys calculator ( hash(nonce+shared_secret) )</string>
         </property>
         <layout class="QGridLayout" name="gridLayout_6">
          <item row="0" column="0">
           <widget class="QLabel" name="label_5">
            <property name="text">
             <string>Nonce</string>
            </property>
           </widget>
          </item>
          <item row="1" column="0" colspan="2">
           <widget class="QLineEdit" name="x25529_textEdit_nonce">
            <property name="font">
             <font>
              <family>Consolas</family>
              <bold>false</bold>
             </font>
            </property>
            <property name="text">
             <string>AA AA AA AA AA AA AA AA AA AA AA AA AA AA AA AA AA AA AA AA AA AA AA AA AA AA AA AA AA AA AA AA</string>
            </property>
            <property name="clearButtonEnabled">
             <bool>true</bool>
            </property>
           </widget>
          </item>
          <item row="2" column="0">
           <widget class="QLabel" name="label_6">
            <property name="text">
             <string>Own AES Key</string>
            </property>
           </widget>
          </item>
          <item row="2" column="1">
           <widget class="QLabel" name="label_7">
            <property name="text">
             <string>Other AES Key</string>
            </property>
           </widget>
          </item>
          <item row="3" column="0">
           <widget class="QLineEdit" name="x25529_textEdit_ownAesKey">
            <property name="font">
             <font>
              <family>Consolas</family>
              <bold>false</bold>
             </font>
            </property>
            <property name="clearButtonEnabled">
             <bool>true</bool>
            </property>
           </widget>
          </item>
          <item row="3" column="1">
           <widget class="QLineEdit" name="x25529_textEdit_otherAesKey">
            <property name="font">
             <font>
              <family>Consolas</family>
              <bold>false</bold>
             </font>
            </property>
            <property name="clearButtonEnabled">
             <bool>true</bool>
            </property>
           </widget>
          </item>
          <item row="4" column="0">
           <widget class="QLabel" name="label_8">
            <property name="text">
             <string>Own AES IV</string>
            </property>
           </widget>
          </item>
          <item row="4" column="1">
           <widget class="QLabel" name="label_9">
            <property name="text">
             <string>Other AES IV</string>
            </property>
           </widget>
          </item>
          <item row="5" column="0">
           <widget class="QLineEdit" name="x25529_textEdit_ownAesIv">
            <property name="font">
             <font>
              <family>Consolas</family>
              <bold>false</bold>
             </font>
            </property>
            <property name="clearButtonEnabled">
             <bool>true</bool>
            </property>
           </widget>
          </item>
          <item row="5" column="1">
           <widget class="QLineEdit" name="x25529_textEdit_otherAesIv">
            <property name="font">
             <font>
              <family>Consolas</family>
              <bold>false</bold>
             </font>
            </property>
            <property name="clearButtonEnabled">
             <bool>true</bool>
            </property>
           </widget>
          </item>
          <item row="6" column="0">
           <widget class="QPushButton" name="x25519_pushButton_CalcSymKeys">
            <property name="text">
             <string>Calculate sym keys</string>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item row="2" column="0">
        <spacer name="verticalSpacer">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>20</width>
           <height>40</height>
          </size>
         </property>
        </spacer>
       </item>
      </layout>
     </widget>
    </widget>
   </item>
  </layout>
 </widget>
 <resources/>
 <connections/>
</ui>
